### PR TITLE
feat: improve position calc and add draggagability to Bezier Editor

### PR DESCRIPTION
**Pull request by roperzh**
Summary of changes:

This refactors the hierarchy used to display the `BezierEditor` popup,
almost all of the changes are made to components used to wrap the Editor
so I don't expect any changes in its behavior.

Things to note:

- We are attaching drag events to `window` instead of attaching them to
the wrapper of the Bezier Editor, this allows us to 1) be more flexible
about the component hierarcy and 2) fix possible edge cases.
- `Popover` in `ui-common` is removed in favor of consolidating all the
logic into `BezierPopup`, in this way we are removing unnecessary tension
product of both components sharing responsibilities.
- Use the `react-draggable` library already present everywhere on the
Timeline to add dnd support to the Editor
- Improve the logic to set the bounds of the Editor.

Regressions to look for:

- Bezier editor positioning

Completed checkin tasks:

- [x] Updated `changelog/public/latest.json`.

### CODE REVIEW

File: packages/haiku-ui-common/src/react/Bezier/BezierEditor.tsx [hunk 1]
```diff
@@ -217,14 +220,16 @@ export default class BezierEditor extends React.Component<BezierEditorProps> {
...
-        };
+    if (readOnly || !down) {
+      window.removeEventListener('mousemove', this.onDownMove);
+      window.removeEventListener('mouseup', this.onDownUp);
+      window.removeEventListener('mouseleave', this.onDownLeave);
+    } else {
+      window.addEventListener('mousemove', this.onDownMove);
+      window.addEventListener('mouseup', this.onDownUp);
+      window.addEventListener('mouseleave', this.onDownLeave);
+    }
```
Comment (stristr) [inline packages/haiku-ui-common/src/react/Bezier/BezierEditor.tsx:245]: qq @roperzh: do we also need to worry about `blur` events here given the webviewness? I am wary of creating stuck drags whenever I see code like this living inside React listeners…there are a few unconventional ways to eject from React and get stuck.

Comment (roperzh) [reply packages/haiku-ui-common/src/react/Bezier/BezierEditor.tsx:245]: @stristr I tried with `cmd+tab`, `cmd+tab+click on icon` (this gave us problems in the past) and by moving/releasing the mouse between webviews and outside of the app window and works fine! do you have other ideas/suggestions in mind?

Comment (stristr) [reply packages/haiku-ui-common/src/react/Bezier/BezierEditor.tsx:245]: Hmmm, probably, but if it's not easy to trigger an error while testing, then we're probably only dealing with edge cases anyway…

Can we at least be sure that we unmount these event listeners when the component unmounts?

**Comment by stristr** [comment]
LGTM.

File: packages/haiku-ui-common/src/react/Bezier/BezierEditor.tsx (hunk 1 above)
Comment (roperzh) [reply packages/haiku-ui-common/src/react/Bezier/BezierEditor.tsx:245]: that's an excellent idea, I pushed that along other improvements.

**Comment by roperzh** [comment]
@stristr I pushed another set of changes, do you want to check again or should I just merge?

**Comment by stristr** [comment]
Go ahead. LGTM still!

**Comment by roperzh** [comment]
Thanks!

**Pull request merged**